manager/integration/tests/test_ha.py

```diff
@@ -2720,11 +2949,37 @@ def test_engine_image_not_fully_deployed_perform_replica_scheduling():
     4. enable the scheduling for node-2
     5. Verify that replicas are scheduled onto node-2 and node-3
     """
-    pass
+    tainted_node_id = \
+        prepare_engine_not_fully_deployed_evnironment(client, core_api)
+
+    # node1: tainted node, node2: self host node, node3: the last one
+    nodes = client.list_node()
+    for node in nodes:
+        if node.id == get_self_host_id():
+            node2 = node
+        elif node.id != tainted_node_id and node.id != node2.id:
+            node3 = node
 
+    node2 = set_node_scheduling(client, node2, allowScheduling=False)
+    node2 = common.wait_for_node_update(client, node2.id, "allowScheduling",
+                                        False)
 
-@pytest.mark.skip(reason="TODO") # NOQA
-def test_engine_image_not_fully_deployed_perform_auto_upgrade_engine():
+    volume1 = create_and_check_volume(client, "vol-1", num_of_replicas=2,
+                                      size=str(3 * Gi))
+
+    volume1.attach(hostId=node3.id)
+    volume1 = common.wait_for_volume_degraded(client, volume1.name)
+
+    node2 = set_node_scheduling(client, node2, allowScheduling=True)
+    node2 = common.wait_for_node_update(client, node2.id,
+                                        "allowScheduling", True)
+
+    volume1 = common.wait_for_volume_healthy(client, volume1.name)
+    for replica in volume1.replicas:
+        assert replica.hostId != tainted_node_id
```

Comment:
I think here need to check if the replicas can be scheduled to node-2 and node-3.